electrum/gui/kivy/uix/dialogs/tx_dialog.py

```diff
@@ -250,6 +253,42 @@ def _do_rbf(
         self.update()
         self.do_sign()
 
+    def do_cpfp(self):
+        from .cpfp_dialog import CPFPDialog
+        parent_tx = self.tx
+        new_tx = self.wallet.cpfp(parent_tx, 0)
+        total_size = parent_tx.estimated_size() + new_tx.estimated_size()
+        parent_txid = parent_tx.txid()
+        assert parent_txid
+        parent_fee = self.wallet.get_tx_fee(parent_txid)
+        if parent_fee is None:
+            self.app.show_error(_("Can't CPFP: unknown fee for parent transaction."))
+            return
+        cb = partial(self._do_cpfp, parent_tx=parent_tx)
+        d = CPFPDialog(self.app, parent_fee, total_size, new_tx=new_tx, callback=cb)
+        d.open()
+
+    def _do_cpfp(
+            self,
+            fee,
+            max_fee,
+            is_final,
+            *,
+            parent_tx: Transaction
+    ):
+        if fee is None:
+            return  # fee left empty, treat is as "cancel"
+        if fee > max_fee:
+            self.show_error(_('Max fee exceeded'))
+            return
+        try:
+            new_tx = self.wallet.cpfp(parent_tx, fee)
+        except CannotCPFP as e:
+            self.app.show_error(str(e))
+            return
+        self.app.tx_dialog(new_tx)
+        self.do_sign()
```

Comment:
`do_sign` does not make sense here as that would be for signing the current tx but that is the parent in this case.
The user can sign the child tx manually using the next tx dialog which the line above opens correctly.

```suggestion
```